### ISSUE TITLE
R1732(consider-using-with): Regression on contextlib.ExitStack

### ISSUE THREAD

**Issue by romainletendart** [PyCQA/pylint#4654]
### Steps to reproduce

Using `exit_stack.enter_context()` within a `with contextlib.ExitStack() as exit_stack` block, I think we shouldn't have any `R1732(consider-using-with)` errors. At least, no errors were raised under `pylint==2.8.3`.

Given a file `test.py`:
```python
import contextlib


def get_serial_number():
    with contextlib.ExitStack() as stack:
        location = stack.enter_context(open('/sys/firmware/devicetree/base/hwid,location', 'r'))
        year = stack.enter_context(open('/sys/firmware/devicetree/base/hwid,year', 'r'))
        week = stack.enter_context(open('/sys/firmware/devicetree/base/hwid,week', 'r'))
        genid = stack.enter_context(open('/sys/firmware/devicetree/base/hwid,genid', 'r'))
        sn = stack.enter_context(open('/sys/firmware/devicetree/base/hwid,sn', 'r'))

        return (
            location.read().rstrip('\x00')
            + year.read()[2:].rstrip('\x00')
            + week.read().rstrip('\x00')
            + genid.read().rstrip('\x00')
            + sn.read().rstrip('\x00')
        )
```

### Current behavior

Result of `pylint test.py`:
```
************* Module test
test.py:6: [R1732(consider-using-with), get_serial_number] Consider using 'with' for resource-allocating operations
test.py:7: [R1732(consider-using-with), get_serial_number] Consider using 'with' for resource-allocating operations
test.py:8: [R1732(consider-using-with), get_serial_number] Consider using 'with' for resource-allocating operations
test.py:9: [R1732(consider-using-with), get_serial_number] Consider using 'with' for resource-allocating operations
test.py:10: [R1732(consider-using-with), get_serial_number] Consider using 'with' for resource-allocating operations

-------------------------------------------------------------------
Your code has been rated at 4.44/10 (previous run: -2.50/10, +6.94)
```

### Expected behavior

```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### pylint --version output

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.8.9 (default, Apr 19 2021, 08:40:21) 
[GCC 10.2.0]
```

The issue is the same with:
```
pylint 2.9.3
astroid 2.6.2
Python 3.8.9 (default, Apr 19 2021, 08:40:21) 
[GCC 10.2.0]
```

However, no issues with:
```
pylint 2.8.3
astroid 2.5.6
Python 3.8.9 (default, Apr 19 2021, 08:40:21) 
[GCC 10.2.0]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I agree with you you should not have a warning here.

**Comment by DudeNr33** [comment]
I will look into this. I was not aware of the ``contextlib.ExitStack.enter_context()`` method.
This should be probably treated in a special way. I wonder if there are any other methods in the standard library that do similar things and should be considered as well.

**Comment by cdce8p** [comment]
> I wonder if there are any other methods in the standard library that do similar things and should be considered as well.

Probably `contextlib.AsyncExitStack.enter_async_context`
https://docs.python.org/3/library/contextlib.html#contextlib.AsyncExitStack.enter_async_context

**Comment by DudeNr33** [comment]
Thanks @cdce8p. 
This led me to realize that the whole ``consider-using-with`` check does not check asnychronous context managers at all.
A quick search over the Github repo of CPython shows that the standard library only has async context managers in form of the various locks of ``asyncio.locks`` and in ``asyncio.events.AbstractServer``.
